GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -260,24 +256,15 @@ - (nullable WKExtension *)sharedApplicationForBackgroundTask {
 #else
 - (nullable id)sharedApplicationForBackgroundTask {
 #endif
-  if ([self isAppExtension]) {
-    return nil;
-  }
-  id sharedApplication = nil;
+  id sharedInstance = nil;
 #if TARGET_OS_IOS || TARGET_OS_TV
-  Class uiApplicationClass = NSClassFromString(@"UIApplication");
-  if (uiApplicationClass &&
-      [uiApplicationClass respondsToSelector:(NSSelectorFromString(@"sharedApplication"))]) {
-    sharedApplication = [uiApplicationClass sharedApplication];
+  if (![self isAppExtension]) {
+    sharedInstance = [UIApplication sharedApplication];
   }
 #elif TARGET_OS_WATCH
-  Class wkExtensionClass = NSClassFromString(@"WKExtension");
-  if (wkExtensionClass &&
-      [wkExtensionClass respondsToSelector:(NSSelectorFromString(@"sharedExtension"))]) {
-    sharedApplication = [wkExtensionClass sharedExtension];
-  }
+  sharedInstance = [WKExtension sharedExtension];
```

Comment:
You might also want to check an extension inside the watch. You can add a todo for now since this is only a future use case for messaging.